server.js

```diff
@@ -1,35 +1,91 @@
+import dotenv from 'dotenv'
+
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import listEndpoints from 'express-list-endpoints'
+
+import booksData from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+dotenv.config()
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/mongo-books"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const bookSchema = new mongoose.Schema({
+  bookID: Number,
+  title: {
+    type: String,
+    lowercase: true
+  },
+  authors: {
+    type: String,
+    uppercase: true
+  },
+  average_rating: Number,
+  num_pages: Number,
+  ratings_count: Number,
+})
+
+const Book = mongoose.model('Book', bookSchema)
+
+if (process.env.RESET_DB) {
+  const seedDataBase = async () => {
+    await Book.deleteMany()
+
+    booksData.forEach(item => {
+      const newBook = new Book(item)
+      newBook.save()
+    })
+  }
+  seedDataBase()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Endpoint that shows all books
+app.get('/books', async (req, res) => {
+  const books = await Book.find(req.query)
+  res.json(books)
+})
+
+// Endpoint that shows books from a specific author (ex: /books?authors=douglas adams)
+app.get('/books', async (req, res) => {
+  const { author } = req.query
+
+  if (author) {
+    const books = await Books.find({
+      author: {
+        $regex: new RegExp(author, "i")
+      }
+    })
+    res.json(books)
+  } else {
+    const books = await Books.find()
+    res.json(books)
+  }
+})
+
+// Endpoint that shows books by id (ex: books/9)
+app.get('/books/:id', async (req, res) => {
+  const { id } = req.params
+  const singleBook = await Book.findOne({ bookID: +id })
```

Comment:
Here it might be useful to specify which ID is being used (bookID), because Mongo also makes its own id, (_id). The latter one would work with Book.findById(id).

Reply:
it is better to use _id field generated by mongo, because it is indexed, that why it would be much faster on bigger datasets.